src/components/SelectionPills/SelectionPill/SelectionPill.d.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+import { WithThemeInjectedProps, ApplyWithTheme } from '../../../styles/themer/withTheme'
+import { RadiumStyles, ElementAttributes } from '../../..'
+
+export interface SelectionPillProps extends WithThemeInjectedProps {
+  /** unique identifier for the pill */
+  id: string | number
+
+  /** Any additional props to add to the list element (e.g. data attributes). */
+  listElementAttributes?: ElementAttributes<'li'>
+
+  /** Any additional props to add to the checkbox element (e.g. data attributes). */
+  elementAttributes?: ElementAttributes<'label'>
+
+  /** Flag determining if component selected state is controlled by parent through props or internal state */
+  parentControlledState?: boolean
+
+  /** Determines wether or not selected styles are applied and start is a selected state */
+  isSelected?: boolean
+
+  /** Determines Whether or not the pill is disabled. */
+  isDisabled?: boolean
+
+  /** Callback function called after pill click */
+  onClick(): void
+
+  /** Callback function called after pill gained focus */
+  onFocus(): void
+
+  /** Callback function called after pill has lost focus */
+  onBlur(): void
```

Comment:
These three need more specific type signatures. The parameters are important for usage, and engineers will not be able to use these callbacks effectively from TS without defining types of the params here.

Also, if these are optional remember to use `?` to mark them as such.